servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieApi.java

```diff
@@ -724,64 +766,89 @@ public void namespaces() throws Exception {
         .containsEntry(ContentKey.of(namespace4.getElements()), namespace4WithId);
 
-    soft.assertThat(api().getNamespace().refName(main.getName()).namespace(namespace1).get())
+    soft.assertThat(api().getNamespace().refName(mainName).namespace(namespace1).get())
         .isEqualTo(namespace1WithId);
-    soft.assertThat(api().getNamespace().refName(main.getName()).namespace(namespace2).get())
+    soft.assertThat(api().getNamespace().refName(mainName).namespace(namespace2).get())
         .isEqualTo(namespace2WithId);
-    soft.assertThat(api().getNamespace().refName(main.getName()).namespace(namespace3).get())
+    soft.assertThat(api().getNamespace().refName(mainName).namespace(namespace3).get())
         .isEqualTo(namespace3WithId);
-    soft.assertThat(api().getNamespace().refName(main.getName()).namespace(namespace4).get())
+    soft.assertThat(api().getNamespace().refName(mainName).namespace(namespace4).get())
         .isEqualTo(namespace4WithId);
 
-    api()
-        .updateProperties()
-        .refName(main.getName())
-        .namespace(namespace2)
-        .updateProperty("foo", "bar")
-        .updateProperty("bar", "baz")
-        .update();
+    if (isV2()) {
+      main = (Branch) api().getReference().refName(mainName).get();
+      UpdateNamespaceResponse update =
+          api()
+              .updateProperties()
+              .refName(mainName)
+              .namespace(namespace2)
+              .updateProperty("foo", "bar")
+              .updateProperty("bar", "baz")
+              .updateWithResponse();
+      soft.assertThat(update.getEffectiveBranch()).isNotNull().isNotEqualTo(main);
+    } else {
+      api()
+          .updateProperties()
+          .refName(mainName)
+          .namespace(namespace2)
+          .updateProperty("foo", "bar")
+          .updateProperty("bar", "baz")
+          .update();
+    }
     Namespace namespace2update =
         (Namespace)
             api()
                 .getContent()
-                .refName(main.getName())
+                .refName(mainName)
                 .key(ContentKey.of("a", "b.b"))
                 .get()
                 .get(ContentKey.of("a", "b.b"));
-    soft.assertThat(api().getNamespace().refName(main.getName()).namespace(namespace2).get())
+    soft.assertThat(api().getNamespace().refName(mainName).namespace(namespace2).get())
         .isEqualTo(namespace2update);
 
     soft.assertThat(
             api()
                 .getMultipleNamespaces()
-                .refName(main.getName())
+                .refName(mainName)
                 .namespace(Namespace.EMPTY)
                 .get()
                 .getNamespaces())
         .containsExactlyInAnyOrder(
             namespace1WithId, namespace2update, namespace3WithId, namespace4WithId);
 
-    api()
-        .updateProperties()
-        .refName(main.getName())
-        .namespace(namespace2)
-        .removeProperty("foo")
-        .update();
+    if (isV2()) {
```

Comment:
those version `if`s look awkward to me... Could we perhaps introduce a utility test method taking a builder and expected v2 data to do the actual API call and validate extra data only in case the API supports it? This way the main test method will be nice and linear... WDYT?

Reply:
The util method could also validate exceptions on the "unsupported" v1 call path.

Reply:
Let's not add unrelated stuff to this PR.